var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
@alalazo you are not entirely correct, one can use perfectly fine `libmkl_rt` as a single dynamic library. As a matter of fact, as of now I can build the whole bunch of things with current MKL like Petsc, Trilinos, deal.II, mumps, etc. I have not tested them, but at least they build and link against MKL.

What you suggest is an option, but requires quite some work. Not sure we gain enough from it.